back-end/app/Http/Controllers/IngredientsController.php

```diff
@@ -1,30 +1,84 @@
-<?php
+<?php declare(strict_types=1);
 
 namespace App\Http\Controllers;
 
+use App\Http\Requests\Ingredient as IngredientReq;
+use App\Repositories\Implementations\Ingredient as IngredientImp;
+use Illuminate\Http\JsonResponse;
+
 class IngredientsController extends Controller
 {
+    /** @var IngredientImp $ingredientImp */
+    private $ingredientImp;
+
+    public function __construct(IngredientImp $ingredientImp)
+    {
+        $this->ingredientImp = $ingredientImp;
+    }
+
     /**
-     * Display a listing of the resource.
+     * Get all Ingredient from DB.
      *
-     * @return \Illuminate\Http\Response
+     * @return JsonResponse
      */
-    public function index()
+    public function index(): JsonResponse
     {
-        return response()->json(['code' =>"Tutti gli ingredienti"],200);
+        $data = $this->ingredientImp->getAll();
 
+        return response()->json(['data' => $data], self::OK);
     }
 
     /**
-     * Display the specified resource.
+     * Get the specified Ingredient from DB.
      *
      * @param  int  $id
-     * @return \Illuminate\Http\Response
+     * @return JsonResponse
+     */
+    public function show(int $id): JsonResponse
+    {
+        $data = $this->ingredientImp->getById($id);
+
+        return response()->json(['data' => $data], self::OK);
+    }
+
+    /**
+     * Method to create an Ingredient.
+     *
+     * @param IngredientReq $request
+     *
+     * @return JsonResponse
      */
-    public function show($id)
+    public function create(IngredientReq $request): JsonResponse
     {
-        return response()->json(['code'=>"Restituzione ingrediente"],200);
+        $name = $request->input('name');
+
+        $ingredients = $this->ingredientImp->create(['name' => $name]);
 
+        if (!empty($ingredients)) {
+            $res =  response()->json(['data' => $ingredients], self::OK);
+        } else {
+            $res = response()->json(['data' => $ingredients], self::NOT_CREATED);
+        }
+
+        return $res;
     }
 
+    /**
+     * Method to delete an Ingredient.
+     *
+     * @param $id
+     *
+     * @return JsonResponse
+     */
+    public function delete(int $id)
+    {
+        $data = $this->ingredientImp->delete($id);
```

Comment:
La `NOT FOUND` andrei sempre a gestire con un eccezione, non con un check su `$data`.